ecmabot.js

```diff
@@ -18,6 +18,7 @@ var JSBot = function(profile) {
 	this.factoids = new FactoidServer(path.join(__dirname, "ecmabot-factoids.json"));
 	this.caniuse_server = new CanIUseServer;
 	this.executeRegex = /^((?:sm?|v8|js?|b|n|>>?|>>>>|\|)>)([^>].*)+/;
+	this.youtubeRegex  = /https?:\/\/www.youtube.com\/watch\?v=(\w+)/;
```

Comment:
i mainly meant the regex :-) you can extract the video ID from both links if needed.